src/components/Choropleth.vue

```diff
@@ -34,43 +34,39 @@ import { APP_BASE_ROUTE } from "../router/utils";
 import { debounce } from "../utils";
 import { backgroundLayer } from "./utils";
 import { useLoadingSpinner } from "../composables/useLoadingSpinner";
+import { useSelectedMapInfo } from "../composables/useSelectedMapInfo";
 
 const mapLoading = ref(true);
 const router = useRouter();
-const { selectedFeatures, selectedIndicators, selectedIndicator, selectedCountryId, appConfig } =
-    storeToRefs(useAppStore());
+const { mapSettings, appConfig } = storeToRefs(useAppStore());
 const featureProperties = appConfig.value.geoJsonFeatureProperties;
 
 const { tooltipForFeature } = useTooltips();
-const { getFillAndOutlineColour } = useColourScale(selectedIndicators);
+const { getFillAndOutlineColour } = useColourScale();
 
 const featureInSelectedCountry = (feature: Feature, selectedCountry: string) =>
     feature.properties[featureProperties.country] === selectedCountry;
 
-const featureAdminLevel = (feature: Feature, selectedCountry) => {
-    return featureInSelectedCountry(feature, selectedCountry) &&
-        !appConfig.value.countriesWithoutAdmin2.includes(selectedCountry)
-        ? 2
-        : 1;
+const getFeatureProp = (feature: Feature, key: "name" | "id") => {
+    const isAdmin2 = featureInSelectedCountry(feature, mapSettings.value.country) && mapSettings.value.adminLevel === 2;
```

Comment:
Is it correct to check whether the feature is in the in the selected country as part of determining whether it's an admin2 feature? I wouldn't expect that behavior from a function whose job is to look up a property of a feature.

Reply:
i think this is in the correct place, only because out of visible features, lets say an admin 2 feature will have props `ID_1`, `ID_2` which are admin level 1 and 2 ids respectively, since the feature itself is an admin level 2 feature, even though it store admin level 1 id, we actually want the admin level 2 id, that is its own id

Reply:
because we have no way of knowing if the feature is an admin 1 or admin 2 feature, however we do have the knowledge that if a country is **not** selected then the feature must be admin level 1 feature, so if it is selected and the state says admin level 2 then we know that this is an admin level 2 feature

Reply:
[I've just clocked it wasn't you who introduced this logic in the first place so sorry for jumping on it!]

I _think_ I follow... though you've just described the proxy we're using as checking "if a country is **not** selected" rather than checking if the feature is "in" the selected country.

So it seems we have two disadvantages here to my mind, one is that the way this logic works is non-obvious, and the other is that we are relying on the 'knowledge' that "if a country is **not** selected then the feature must be admin level 1 feature". Since that isn't a law of physics it's possible that it won't always be true. For example, maybe at some point we introduce an admin0-level view; or something changes that means that the `featureInSelectedCountry` check is no longer a totally accurate proxy for 'a country is selected' (could it happen around disputed regions?)

An attempt to short-cut the two disadvantages: check whether the feature is admin2 by seeing whether it has the `featureProperties.idAdm2` key. (This scales nicely if we introduce admin level 3!)

Reply:
i can potentially see this working, although havent thought about all the implications of this, happy to create a ticket about this because i think your idea will work!

Reply:
If _that_ doesn't work I would advocate for a comment on how checking presence of country is required to diagnose the admin level, eg:
```
// There is no way to tell if a feature's level is admin1 or admin2 by looking at it. But, if no country is selected, we can infer that the level is admin1.
```

Reply:
yep made a ticket that links to this discussion! https://mrc-ide.myjetbrains.com/youtrack/issue/mrc-5376/Improve-way-we-get-feature-properties